.github/workflows/ci.yaml

```diff
@@ -0,0 +1,108 @@
+name: CI Pipeline
+
+on:
+  workflow_call:
+    secrets:
+      MAIL_USERNAME:
+        description: 'Username for the email notification step'
+        required: true
+      MAIL_PASSWORD:
+        description: 'Password for the email notification step'
+        required: true
+    inputs:
+      custom_docker_tag:
+        description: 'Custom Docker tag'
+        # If not specified, then the 'latest' tag will be used
+        default: ''
+        required: false
+        type: string
+
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  DOCKER_REGISTRY: ghcr.io
+  DOCKER_REPO: angular-react-starter-image
+  DOCKER_TAG: ${{ inputs.custom_docker_tag || 'latest' }}
+  MAIL_SERVER_ADDRESS: smtp.gmail.com
+  MAIL_SERVER_PORT: 465
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./angular
+    steps:
+      - name: Checkout code from the repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Install Clean dependencies
+        run: npm ci
+
+      - name: Run headless tests
+        id: run-tests
+        if: ${{ always() }}
+        run: npm run test -- --watch=false --browsers=ChromeHeadless 
+
+      - name: Build Angular project
+        id: build-project
+        if: steps.run-tests.outcome == 'success'
+        run: npm run build
+
+      - name: Log in to GitHub Container Registry  
+        id: login-registry
+        if: steps.build-project.outcome == 'success'
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login $DOCKER_REGISTRY -u ${{ github.actor }} --password-stdin
+
+      - name: Build Docker image
+        id: build-docker
+        if: steps.login-registry.outcome == 'success'
+        run: docker build -t $DOCKER_REGISTRY/${{ github.actor }}/$DOCKER_REPO:$DOCKER_TAG .
+        working-directory: .
+
+      - name: Push Docker image
+        id: push-docker
+        if: steps.build-docker.outcome == 'success'
+        run: docker push $DOCKER_REGISTRY/${{ github.actor }}/$DOCKER_REPO:$DOCKER_TAG
+
+      - name: Get pipeline execution time
+        id: get-time
+        if: ${{ always() }}
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: echo "execution_time=$( gh run list --workflow ci.yaml --limit 1 | grep -E --only-matching '([0-9]{1,2}[hms]){1,3}' )" >> $GITHUB_ENV
+
+      - name: Send email notification
+        if: ${{ always() }}
+        uses: dawidd6/action-send-mail@v3
+        with:
+          server_address: ${{ env.MAIL_SERVER_ADDRESS }}
+          server_port: $MAIL_SERVER_PORT
+
+          username: ${{ secrets.MAIL_USERNAME }}
+          password: ${{ secrets.MAIL_PASSWORD }}
+
+          subject: GITHUB ACTIONS JOB RESULT
+
+          to: "${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com"
```

Comment:
Let's move this `to` email to a var too, if nothing is passed it would be calculated in a way like it is now

Reply:
Done